### PR TITLE
refactor(web): byte unit utils

### PR DESCRIPTION
**Pull request by danieldietzler**
Simplify byte utils using ✨ Math ✨

### CODE REVIEW

**Review by mertalev — APPROVED** [review]

File: web/src/lib/utils/byte-units.ts [hunk 1]
```diff
@@ -8,23 +18,10 @@
...
-      break;
-    }
-  }
-
-  remainder = Number.parseFloat(remainder.toFixed(maxPrecision));
+export function getBytesWithUnit(bytes: number, maxPrecision = 1): [number, ByteUnit] {
+  const magnitude = Math.floor(Math.log2(bytes === 0 ? 1 : bytes) / 10);
 
-  return [remainder, units[magnitude]];
+  return [Number.parseFloat((bytes / 2 ** (magnitude * 10)).toFixed(maxPrecision)), magnitude];
```
Comment (michelheusschen) [inline web/src/lib/utils/byte-units.ts:24]: Maybe we can use base 1024 for these functions? But this isn't wrong at all.

```suggestion
  const magnitude = Math.floor(Math.log(bytes === 0 ? 1 : bytes) / Math.log(1024));

  return [Number.parseFloat((bytes / 1024 ** magnitude).toFixed(maxPrecision)), magnitude];
```

**Review by michelheusschen — APPROVED** [review]
Nice!

File: web/src/lib/utils/byte-units.ts (hunk 1 above)
Comment (danieldietzler) [reply web/src/lib/utils/byte-units.ts:24]: Oh yeah we could, good idea!

**Review by jrasm91 — APPROVED** [review]

**Comment by aviv926** [comment]
Could this change affect https://github.com/immich-app/immich/issues/10927?

**Pull request merged**